src/libraries/System.Linq/src/System/Linq/SkipTake.SpeedOpt.cs

```diff
@@ -148,6 +148,12 @@ public void CopyTo(TSource[] array, int arrayIndex) =>
 
             private static void Fill(IList<TSource> source, Span<TSource> destination, int sourceIndex)
             {
+                if (source.TryGetSpan(out ReadOnlySpan<TSource> sourceSpan))
+                {
+                    sourceSpan.Slice(sourceIndex, destination.Length).CopyTo(destination);
+                    return;
+                }
+
```

Comment:
If we're updating this, we should also update IndexOf to delegate to MemoryExtensions.IndexOf when we can get the span. The main benefit of that is it'll then be vectorized when TSource is an appropriate type.